### PR TITLE
Adds lentils

### PR DESCRIPTION
**Pull request by aoae23**
#### Summary
```SUMMARY: Content "Adds lentils"```  

#### Purpose of change
Lentils are a common food consumed around the world and, while admittedly grown infrequently in the northeast, are grown across the world. (Then again chili peppers can be grown in this game.) This PR adds lentils (both raw and dried) and cooked lentils, and adds lentils to curry and soup recipes. I sourced the nutrition data mainly from https://ndb.nal.usda.gov/ndb/search/list.

#### Describe alternatives you've considered
Not adding lentils.

Adding canned lentils and other lentil-containing items would be a good next step if this PR works out.

#### Additional context
Thanks to everyone in the C:DDA Discord for walking me through the procedure.

Please let this work. I did my best to avoid bugs but there may be some mistakes in there.

### CODE REVIEW

File: data/json/items/comestibles/other.json [hunk 1]
```diff
@@ -306,6 +306,44 @@
     "vitamins": [ [ "calcium", 2 ], [ "vitB", 8 ] ],
     "rot_spawn": "GROUP_ROTTING_PLANT"
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "dry_lentil",
+    "name": "dried lentils",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/other.json]: Please add a plural name here too.

File: data/json/item_groups.json [hunk 2]
```diff
@@ -583,6 +583,7 @@
       ["soap", 70],
       ["detergent", 50],
       ["dry_beans", 40],
+      ["dry_lentils", 30],
```
Comment (I-am-Erk) [inline data/json/item_groups.json]: This and the other similar occurences need to be formatted [ "dry_lentils", 30 ], to pass Jenkins. It's a format standard thing. Don't worry about the outdated formatting on the other entries in the file, Jenkins will ignore them.

Comment (anothersimulacrum) [reply data/json/item_groups.json]: The JSON linter can be found [here](http://dev.narc.ro/cataclysm/format.html).

Comment (I-am-Erk) [reply data/json/item_groups.json]: It can be, but this file should not be run through the linter.

Comment (aoae23) [reply data/json/item_groups.json]: Fixed the formatting.

**Pull request merged**